framework/wazuh/core/stats.py

```diff
@@ -2,14 +2,207 @@
 # Created by Wazuh, Inc. <info@wazuh.com>.
 # This program is free software; you can redistribute it and/or modify it under the terms of GPLv2
 
+import configparser
 import json
 import os
 from datetime import datetime
+from io import StringIO
 
-from wazuh.core import common
-from wazuh.core.exception import WazuhInternalError, WazuhError
+from wazuh.core import agent, common
+from wazuh.core.exception import (WazuhError, WazuhException,
+                                  WazuhInternalError, WazuhResourceNotFound)
 from wazuh.core.wazuh_socket import WazuhSocket
 
+DAYS = "Sun", "Mon", "Tue", "Wed", "Thu", "Fri", "Sat"
+MONTHS = "Jan", "Feb", "Mar", "Apr", "May", "Jun", "Jul", "Aug", "Sep", "Oct", "Nov", "Dec"
+
+
+def hourly_():
+    """Compute hourly averages
+
+    Returns
+    -------
+    array
+        averages and iterations
+    """
+    averages = []
+    interactions = 0
+    for i in range(25):
+        try:
+            hfile = open(common.stats_path + '/hourly-average/' + str(i))
+            data = hfile.read()
+            if i == 24:
+                interactions = int(data)
+            else:
+                averages.append(int(data))
+            hfile.close()
+        except IOError:
+            if i < 24:
+                averages.append(0)
+
+    return [{'averages': averages, 'interactions': interactions}]
+
+
+def weekly_():
+    """Compute weekly averages
+
+    Returns
+    -------
+    array
+        hours and interactions for each week day
+    """
+    weekly_results = []
+    for i in range(7):
+        hours = []
+        interactions = 0
+        for j in range(25):
+            try:
+                wfile = open(common.stats_path + '/weekly-average/' + str(i) + '/' + str(j))
+                data = wfile.read()
+                if j == 24:
+                    interactions = int(data)
+                else:
+                    hours.append(int(data))
+                wfile.close()
+            except IOError:
+                if i < 24:
+                    hours.append(0)
+        weekly_results.append({DAYS[i]: {'hours': hours, 'interactions': interactions}})
+
+    return weekly_results
+
+
+def totals_(date):
+    """Statistical information for the current or specified date
+
+    Parameters
+    ----------
+    date : date
+        date object with the date value of the stats
+
+    Returns
+    -------
+    tuple
+        failed: boolean that represents if data parsing failed
+        affected: array of dictionaries. Each dictionary represents an hour
+
+    Raises
+    ------
+    WazuhError
+        raised on IOError
+    """
```

Comment:
```suggestion
    """Statistical information for the current or specified date.

    Parameters
    ----------
    date: date
        Date object with the date value of the stats.

    Returns
    -------
    tuple
        Failed: boolean that represents if data parsing failed. Its value is always `False`.
        affected: array of dictionaries. Each dictionary represents an hour.

    Raises
    ------
    WazuhError
        Raised on `IOError`.
    """
```

Review all the docstrings to apply the following format.